### DISCUSSION TITLE
how the Topic Probabilities for each document are estimated in BERTopic?

### DISCUSSION THREAD

**Discussion by chonchonj23** [MaartenGr/BERTopic#1467] [Q&A]
Dear Maarten, 

I was wondering how the probabilities (generated in the fit_transform step) are calculated behind the scene.

topics, probs = model.fit_transform(docs)

Thank you very much in advance. 

Best regards,
Chon

**Comment by MaartenGr** [comment]
The probabilities are generated through the underlying clustering algorithm, namely HDBSCAN. You can find more information about that in the [documentation](https://maartengr.github.io/BERTopic/algorithm/algorithm.html#visual-overview) as well as [here](https://hdbscan.readthedocs.io/en/latest/soft_clustering_explanation.html).
